### PR TITLE
Encapsulation of Target related functionality

### PR DESCRIPTION
**Pull request by mike-sul**
Initial effort to encapsulate Target-related functionality.

### CODE REVIEW

**Review by doanac — COMMENTED** [review]
ignoring the first 2 commits that are in other PRs:

I believe this is just reshuffling code, right? It took a minute, but I don't think anything is changing functionally?

File: src/liteclient.cc [hunk 1]
```diff
@@ -118,6 +122,82 @@ LiteClient::LiteClient(Config& config_in)
...
+      }
+    }
+  }
+  if (find_latest && latest != nullptr) {
+    return latest;
+  }
+  throw std::runtime_error("Unable to find update");
+}
+
+boost::container::flat_map<int, Uptane::Target> LiteClient::getTargets() {
```
[the remark this answers is not in the document]
Comment (mike-sul) [reply src/liteclient.cc:177]: > So this encapsulating some old code that's producing a sorted list of applicable targets?
yes, it was in `list_main()`, effectively, it does quite similar to what getTarget() does, so I will unify them at some point.

I am not sure about renaming, I would keep the class method name as simple as possible.
LiteClient returns to its clients only available/applicable Targets by definition, there is no point in returning of any non-related Targets, thus no point to inject 'applicable' into the method name.
The return type (`boost::container::flat_map`) implies that the resultant map is sorted/ordered.

I think it makes sense to add a comment to the method declaration in the header file explaining the details you mentioned.

**Comment by mike-sul** [comment]
> I believe this is just reshuffling code, right? It took a minute, but I don't think anything is changing functionally?

yes, that's correct.

File: src/main.cc [hunk 2]
```diff
@@ -60,78 +60,15 @@ static int list_main(LiteClient& client, const bpo::variables_map& unused) {
...
-      if (it == hwid) {
-        sorted_targets.emplace(ver, t);
-        break;
-      }
-    }
-  }
+  boost::container::flat_map<int, Uptane::Target> sorted_targets = client.getTargets();
 
-  LOG_INFO << "Updates available to " << hwid << ":";
+  LOG_INFO << "Availbale updates: ";
```
[the remark this answers is not in the document]
Comment (mike-sul) [reply src/main.cc]: fixed, force updated.

File: src/liteclient.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (mike-sul) [reply src/liteclient.cc:177]: added the comment

File: src/main.cc [hunk 3]
```diff
@@ -195,8 +194,7 @@ static int daemon_main(LiteClient& client, const bpo::variables_map& variables_m
         std::string reason = "Updating from " + current.filename() + " to " + found_latest_target->filename();
```
Comment (doanac) [inline src/main.cc:194]: Not sure github is including the context for my comment: We have this line:
~~~
if (!known_target_sha && !client.isTargetActive(*found_latest_target)) {
~~~

Its calling `getCurrent` under the covers. I wonder if we should drop that and roughly do
~~~
if (!known_target_sha && current != found_latest_target) {
~~~

File: src/composeappmanager.cc [hunk 4]
```diff
@@ -67,7 +67,7 @@ ComposeAppManager::ComposeAppManager(const PackageConfig& pconfig, const Bootloa
     LOG_DEBUG << "Failed to initialize Compose App Tree (ostree) at " << cfg_.apps_tree << ". Error: " << exc.what();
   }
 
-  const auto& current_apps = getApps(getCurrent());
+  const auto& current_apps = getApps(OstreeManager::getCurrent());
```
Comment (doanac) [inline src/composeappmanager.cc:70]: Is the idea here that once `ComposeAppManager::getCurrent` handles the "TODO", all we really need from this method is the logic of its parent class?

**Pull request closed without merging**